src/globalEndpointManager.ts

```diff
@@ -95,6 +95,10 @@ export class GlobalEndpointManager {
     return this.locationCache.resolveServiceEndpoint(request);
   }
 
+  public async setServiceEndpoint(request: RequestContext) {
```

Comment:
nit: I'd prefer something that's functional. 

Also not ideal from a perf perspective since we're calling an extra await call here now.

Reply:
Removed this